src/coreclr/tools/r2rdump/CoreDisTools.cs

```diff
@@ -1209,6 +1213,235 @@ private static bool IsAnotherRuntimeFunctionWithinMethod(int rva, RuntimeFunctio
+
+                if (isFound)
+                {
+                    if (!TryGetImportCellName(target + immValue, out string targetName) || string.IsNullOrWhiteSpace(targetName))
+                    {
+                        return;
+                    }
+
+                    StringBuilder updatedInstruction = new();
+                    updatedInstruction.Append(instruction);
+                    updatedInstruction.Append(" // ");
+                    updatedInstruction.Append(targetName);
+                    instruction = updatedInstruction.ToString();
+                }
+            }
+
+        }
+
+        /// <summary>
+        /// Counts the value stored in a given register.
+        /// </summary>
+        /// <param name="currentPC">Program Counter of analyzed instruction</param>
+        /// <param name="currentInstrOffset">Offset within the runtime function of analyzed instruction</param>
+        /// <param name="imageOffset">Offset within the image byte array</param>
+        /// <param name="register">Register whose value is being searched for</param>
+        /// <param name="registerValue">Found value of the register</param>
+        /// <returns>If the value stored in a given register is countable, the function returns true. Otherwise false</returns>
+        private bool StaticAnalyzeRiscV64Assembly(int currentPC, int currentInstrOffset, int imageOffset, uint register, ref int registerValue)
+        {
+            const int InstructionSize = 4;
+
+            // if currentInstrOffset is less than 0 then it is the end of this analyzed method
+            if (currentInstrOffset < 0)
+            {
+                return false;
+            }
+
+            do
+            {
+                uint instr = BitConverter.ToUInt32(_reader.Image, imageOffset + currentInstrOffset);
+                if (IsRiscV64AddiInstruction(instr))
+                {
+                    AnalyzeRiscV64Itype(instr, out uint rd, out uint rs1, out int imm);
+                    if (rd == register)
+                    {
+                        int rs1Value = 0;
+                        bool returnValue = StaticAnalyzeRiscV64Assembly(currentPC - InstructionSize, currentInstrOffset - InstructionSize, imageOffset, rs1, ref rs1Value);
+                        registerValue = rs1Value + imm;
+                        return returnValue;
+                    }
+                }
+                else if (IsRiscV64AuipcInstruction(instr))
+                {
+                    AnalyzeRiscV64Utype(instr, out uint rd, out int imm);
+                    if (rd == register)
+                    {
+                        registerValue = currentPC + imm;
+                        return true;
+                    }
+                }
+                else
+                {
+                    // check if "register" is calculated using an unsupported instruction
+                    uint rd = (instr >> 7) & 0b_11111U;
```

Comment:
If instr is not IType like `sd` which 11-7 bits are just offset. It can return false when immediate value is same to register. I think this pattern is not produced in our runtime. I just want to check it is intended?
Thank you.

Reply:
In my honest opinion, i have never seen a situation when an instruction, like `sd`, would broke the pattern which is acceptable by `ProbeRiscV64Quirks(...)`. So I assume that it is not produced in the runtime.